Source/player.h

```diff
@@ -8,185 +8,223 @@
+	PM_BLOCK,
+	PM_GOTHIT,
+	PM_DEATH,
+	PM_SPELL,
+	PM_NEWLVL,
+	PM_QUIT,
+} PLR_MODE;
+
+typedef enum action_id {
+	// clang-format off
+	ACTION_WALK        = -2, // Automatic walk when using gamepad
+	ACTION_NONE        = -1,
+	ACTION_ATTACK      = 9,
+	ACTION_RATTACK     = 10,
+	ACTION_SPELL       = 12,
+	ACTION_OPERATE     = 13,
+	ACTION_DISARM      = 14,
+	ACTION_PICKUPITEM  = 15, // put item in hand (inventory screen open)
+	ACTION_PICKUPAITEM = 16, // put item in inventory
+	ACTION_TALK        = 17,
+	ACTION_OPERATETK   = 18, // operate via telekinesis
+	ACTION_ATTACKMON   = 20,
+	ACTION_ATTACKPLR   = 21,
+	ACTION_RATTACKMON  = 22,
+	ACTION_RATTACKPLR  = 23,
+	ACTION_SPELLMON    = 24,
+	ACTION_SPELLPLR    = 25,
+	ACTION_SPELLWALL   = 26,
+	// clang-format on
+} action_id;
+
+typedef enum player_weapon_type {
+	WT_MELEE,
+	WT_RANGED,
+} player_weapon_type;
+
 typedef struct PlayerStruct {
-	int _pmode;
-	char walkpath[MAX_PATH_LENGTH];
-	BOOLEAN plractive;
-	int destAction;
-	int destParam1;
-	int destParam2;
-	int destParam3;
-	int destParam4;
-	int plrlevel;
-	int _px;      // Tile X-position of player
-	int _py;      // Tile Y-position of player
-	int _pfutx;   // Future tile X-position of player. Set at start of walking animation
-	int _pfuty;   // Future tile Y-position of player. Set at start of walking animation
-	int _ptargx;  // Target tile X-position for player movment. Set during pathfinding
-	int _ptargy;  // Target tile Y-position for player movment. Set during pathfinding
-	int _pownerx; // Tile X-position of player. Set via network on player input
-	int _pownery; // Tile X-position of player. Set via network on player input
-	int _poldx;   // Most recent X-position in dPlayer.
-	int _poldy;   // Most recent Y-position in dPlayer.
-	int _pxoff;   // Player sprite's pixel X-offset from tile.
-	int _pyoff;   // Player sprite's pixel Y-offset from tile.
-	int _pxvel;   // Pixel X-velocity while walking. Indirectly applied to _pxoff via _pvar6
-	int _pyvel;   // Pixel Y-velocity while walking. Indirectly applied to _pyoff via _pvar7
-	int _pdir;    // Direction faced by player (direction enum)
-	int _pgfxnum; // Bitmask indicating what variant of the sprite the player is using. Lower byte define weapon (anim_weapon_id) and higher values define armour (starting with anim_armor_id)
-	unsigned char *_pAnimData;
-	int _pAnimDelay; // Tick length of each frame in the current animation
-	int _pAnimCnt;   // Increases by one each game tick, counting how close we are to _pAnimDelay
-	int _pAnimLen;   // Number of frames in current animation
-	int _pAnimFrame; // Current frame of animation.
-	int _pAnimWidth;
-	int _pAnimWidth2;
-	int _peflag;
-	int _plid;
-	int _pvid;
-	int _pSpell;
-	char _pSplType;
-	char _pSplFrom;
-	int _pTSpell;
-	char _pTSplType;
-	int _pRSpell;
+	PLR_MODE _pmode;
+	Sint8 walkpath[MAX_PATH_LENGTH];
+	bool plractive;
+	action_id destAction;
+	Sint32 destParam1;
+	Sint32 destParam2;
+	direction destParam3;

```

Comment:
Sure its safe to type this one? is it really always a direction?

Reply:
yep.